relentless/ensemble.py

```diff
@@ -1,176 +1,437 @@
 
         return self
 
     def copy(self):
-        ens = copy.deepcopy(self)
-        return ens.reset()
-
-    def save(self, basename='ensemble'):
-        # dump thermo data to json file
-        data = {'types': self.types,
-                'kB': self.kB,
-                'T': self.T,
-                'P': self.P,
-                'V': self.V,
-                'N': self.N,
-                'mu': self.mu,
-                'conjugates': self.conjugates}
-        with open('{}.json'.format(basename),'w') as f:
-            json.dump(data, f, sort_keys=True, indent=4)
-
-        # dump rdfs in separate files
-        for pair in self.rdf:
-            if self.rdf[pair] is not None:
-                i,j = pair
-                np.savetxt('{}.{}.{}.dat'.format(basename,i,j), self.rdf[pair].table, header='r g[{},{}](r)'.format(i,j))
+        """Makes a copy of the ensemble.
+
+        Returns
+        -------
+        :py:class:`Ensemble`
+            A copy of the ensemble object.
+
+        """
+        return copy.deepcopy(self)
+
+    def save(self, filename):
+        """Saves the ensemble parameters and the RDF values into a JSON file.
+
+        Parameters
+        ----------
+        filename : `str`
+            The name of the file to save data in.
+
+        """
+        data = []
+
+        # append thermo data
+        thermo = {'T': self.T,
+                  'P': self.P,
+                  'V': self.V,
+                  'mu':self.mu,
+                  'N': self.N,
+                  'kB': self.kB,
+                  'constant': self.constant} #need constant?
+        data.append(thermo)
+
+        # append rdf data
+        rdf = {pair:self.rdf[pair].table for pair in self.rdf}
+        data.append(rdf)
+
+        # dump data to json file
+        with open('{}.json'.format(filename),'w') as f:
+            json.dump(data, f, indent=4)
 
     @classmethod
-    def load(self, basename='ensemble'):
-        with open('{}.json'.format(basename)) as f:
+    def load(self, filename):
+        """Loads ensemble parameters and RDF values from a JSON file into a new
+        :py:class:`Ensemble` object.
+
+        Parameters
+        ----------
+        filename : `str`
+            The name of the file from which to load data.
+
+        Returns
+        -------
+        :py:class:`Ensemble`
+            An ensemble object with parameter values from the JSON files.
+
+        """
+        with open('{}.json'.format(filename)) as f:
             data = json.load(f)
 
-        ens = Ensemble(types=data['types'],
-                       T=data['T'],
-                       P=data['P'],
-                       V=data['V'],
-                       N=data['N'],
-                       mu=data['mu'],
-                       kB=data['kB'],
-                       conjugates=data['conjugates'])
+        thermo = data[0]
+        rdf = data[1]
+
+        #reset fluctuating variables?
+
+        ens = Ensemble(T=thermo['T'],
```

Comment:
This probably won't work because you might have both constant and fluctuating variables saved in the file. I think you will need to go through `constants` to figure out what the constructor arguments are and use those to make the ensemble, then fill in the fluctuating ones.